### PR TITLE
add linear algebra and documentation

### PR DESCRIPTION
**Pull request by apkille**
- Adds the following linear algebra functionalities:
    - Complex conjugate `conj`
    - Transpose `transpose`
    - Exponential of an operator `exp`
    - Vectorization of an operator `vec`
- Adds a guide documentation page for getting started with the package, now that we have extensive functionalities for creating, manipulation, and numerically translating expressions.
- Cleans up and adds docstrings.

### CODE REVIEW

File: docs/src/introduction.md [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+julia> @op C SpinBasis(1//2)⊗SpinBasis(5//2)
+C
+
+julia> basis(C)
+[Spin(1/2) ⊗ Spin(5/2)]
+```
+Here, we extracted the basis of the defined symbolic operators using the `basis` function.
+
+
+Symbolic quantum objects with additional properties can be defined, such as a Hermitian operator, or the zero ket (i.e., a symbolic ket equivalent to the zero vector $\bm{0}$).
```
Comment (Krastanov) [inline docs/src/introduction.md:55]: Did you have a chance to render the docs locally to check if this latex expression is rendered correctly?

You can run the local compilation with 
```
julia --project=docs -e '
            using Pkg
            Pkg.develop(PackageSpec(path=pwd()))
            Pkg.instantiate()';
julia --project=docs -i docs/make.jl
```

File: docs/src/introduction.md [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+- transpose: `transpose`,
+- projection of a ket: `projector`,
+- adjoint or dagger: `dagger`,
+- inverse of an operator: `inv`,
+- exponential of an operator: `exp`,
+- vectorization of an operator: `vec`.
+
+## Simplifying Expressions
+
+For predefined objects such as the Pauli operators `X`, `Y`, and `Z`, manual simplification can be performed with the [`qsimplify`](@ref) function. Take the following example:
```
Comment (Krastanov) [inline docs/src/introduction.md:172]:
```suggestion
For predefined objects such as the Pauli operators `X`, `Y`, and `Z`, additional simplification can be performed with the [`qsimplify`](@ref) function. Take the following example:
```

**Review by Krastanov — CHANGES_REQUESTED** [review]
This looks good! Left some very minor comments that you can disregard if you wish or if you have different plans or styles in mind. Feel free to merge

File: docs/src/introduction.md (hunk 1 above)
Comment (apkille) [reply docs/src/introduction.md:55]: Yes I did, it renders correctly.

**Pull request merged**